frontend/src/main/java/starter/project/frontend/CarController.java

```diff
@@ -13,23 +13,19 @@
 
     @PostMapping(path = "/car", consumes = {"application/json"})
     public Map<Integer,Car> addCar(@RequestBody Car car) {
-        map.put(map.size()+1,car);
-        car.setId(map.size());
+
+        map.put(car.getId(),car);
+
         return map;
     }
 
     @GetMapping(path = "/car/{carId}")
     public Car getCar(@PathVariable("carId") int carId) {
-        for (Map.Entry<Integer, Car>  entry : map.entrySet()) {
-            int key = entry.getKey();
-            Car car = entry.getValue();
-            if (car.getId() == (carId)) {
+
+     Car car =  map.get(carId);
+
                 return car;
-            }
         }
-        throw new ResponseStatusException(
-                HttpStatus.NOT_FOUND, "entity not found");
-    }
 
     @GetMapping(path = "/cars/{list}")
     public List<Car> getBatchCars(@PathVariable List<Integer> list) {
```

Comment:
This endpoint doesn't work. We need to parse the list pathVariable somehow or include the list of ids in the request body instead of in the path. At the moment we can't get a `List<Integer>` from `{list}` (FWIW, you can but it will always be size 1)

Reply:
GET requests shouldn't have a request body.
This works in spring magically. The request should look like `/cars/1,2,3,4`

Reply:
TIL that's a thing. Apologies, I'll mark this as resolved